src/backend/distributed/planner/recursive_planning.c

```diff
@@ -379,6 +362,27 @@ RecursivelyPlanSubqueriesAndCTEs(Query *query, RecursivePlanningContext *context
 													 query, context);
 	}
 
+	/*
```

Comment:
just noting for future reference & documentation purposes: It is fine to move this check after `RecursivelyPlanNonColocatedSubqueries` as well. Because, `RecursivelyPlanNonColocatedSubqueries` does explicitly pick a "distributed" range table entry, and makes sure that it survives through `RecursivelyPlanNonColocatedSubqueries`. So, we cannot end up with a query where `RecursivelyPlanNonColocatedSubqueries` makes the FROM clause recurring and then   `ShouldRecursivelyPlanSublinks` kicks in.